src/python/bcc/table.py

```diff
@@ -397,6 +397,34 @@ def clear(self):
         for k in self.keys():
             self.__delitem__(k)
 
+    def items_lookup_and_delete_batch(self, batch_size=0):
+        # By default batch size is the biggest possible (max_entries in
+        # the map) unless user wants to override it.
+        if batch_size <= 0:  # no value set by user
+            batch_size = self.max_entries
+
+        in_batch = ct.c_void_p(0)
+        out_batch = ct.c_void_p(0)
```

Comment:
So, in_batch and out_batch are both pointers to `0` right? The `batch` is an opaque number used for continuous batch processing in the case of smaller buffer. For example:
  1. first run, in_batch = NULL, out_patch content not matter
  2. second run, in_batch = previous out_patch, out patch content not matter,
  3. ...
Since here, we allocate enough buffer and only want to do one batch processing,
the in_batch needs to be a NULL pointer.

Reply:
Yes this is inspired by what I have seen in /kernel-src/tools/testing/selftests/bpf/map_tests/htab_map_batch_ops.c

Your comment make me change my mind about the optional argument : batch_size. It does not do what we could intuitively imagine. It does a lookup_and_delete on only batch_size elements in the table then returns. I does not loop until the map is empty and returns batch_size elements on each steps.
I think this is confusing and not useful. What do you think about it ?

Reply:
> Yes this is inspired by what I have seen in /kernel-src/tools/testing/selftests/bpf/map_tests/htab_map_batch_ops.c
> 
> Your comment make me change my mind about the optional argument : batch_size. It does not do what we could intuitively imagine. It does a lookup_and_delete on only batch_size elements in the table then returns. I does not loop until the map is empty and returns batch_size elements on each steps.
> I think this is confusing and not useful. What do you think about it ?

Yes, let us just remove the optional argument and users in general do
not really care about batch size. If there is a need for map with
*huge* size, we can improve to using smaller batch size inside
the API.